rust/azure_iot_operations_mqtt/src/connection_settings.rs

```diff
@@ -330,140 +338,165 @@ fn string_from_environment(key: &str) -> Result<Option<String>, String> {
     match env::var(key) {
         Ok(value) => Ok(Some(value)),
         Err(VarError::NotPresent) => Ok(None), // Handled by the validate function if required
-        Err(e) => Err(format!("Parsing {key} from environment failed: {e}")),
+        Err(VarError::NotUnicode(_)) => {
+            Err("Could not parse non-unicode environment variable".to_string())
+        }
+    }
+}
+
+/// Helper function to extract the value from a config map file as a string.
+fn string_from_configmap_file(
+    configmap_path: &Path,
+    filename: &str,
+) -> Result<Option<String>, String> {
+    let file_pathbuf = configmap_path.join(filename);
+    if !file_pathbuf.as_path().exists() {
+        return Ok(None); // File doesn't exist, return None
     }
+    let value = std::fs::read_to_string(file_pathbuf)
+        .map_err(|e| format!("Malformed {filename} file: {e}"))?;
+    // NOTE: Can't chain .trim() due to memory rules
+    Ok(Some(value.trim().to_string()))
 }
 
 #[cfg(test)]
 mod tests {
-    use super::MqttConnectionSettingsBuilder;
-    // use std::env;
-    // use std::fs;
-    // use std::path::PathBuf;
-    // use std::sync::Mutex;
+    use super::*;
+    use std::fs;
+    use std::path::Path;
+    use test_case::test_case;
+
+    /// Struct that managed a temporary Config Map created for testing purposes.
+    /// When dropped, the Config Map directory, and all files in it will be removed.
+    struct TempConfigMapManager {
+        dir: tempfile::TempDir,
+    }
+
+    impl TempConfigMapManager {
+        fn new(dir_name: &str) -> Self {
+            Self {
+                dir: tempfile::TempDir::with_prefix(dir_name).unwrap(),
+            }
+        }
 
-    //pub static FILE_DIR_MTX: Mutex<()> = Mutex::new(()); // TODO: Find a better way to control test flow
+        fn add_file(&self, file_name: &str, contents: &str) {
+            let file_path = self.dir.path().join(file_name);
+            fs::write(file_path, contents).unwrap();
+        }
+
+        fn path(&self) -> &Path {
+            self.dir.path()
+        }
+    }
 
     #[test]
-    fn test_connection_settings_empty_hostname() {
+    fn minimum_configuration() {
```

Comment:
I think we usually start test functions with `test_` to help distinguish them from test helper functions

Reply:
Yeah, I'm trying to move away from that - it's very redundant in the test output. The macro on the top should be sufficient.